tests/testthat/test-filter_confirmation.R

```diff
@@ -0,0 +1,142 @@
+  actual <-
+    filter_confirmation(
+      data,
+      by_vars = vars(USUBJID),
+      join_vars = vars(AVISITN, AVALC),
+      order = vars(AVISITN),
+      filter = AVALC == "PR" & AVALC.join %in% c("CR", "PR") &
+        AVISITN < AVISITN.join
+    )
+
+  expected <- tibble::tribble(
+    ~USUBJID, ~AVISITN, ~AVALC,
+    "1",      1,        "PR",
+    "4",      1,        "PR"
+  )
+
+  expect_dfs_equal(
+    base = expected,
+    compare = actual,
+    keys = c("USUBJID", "AVISITN")
+  )
+})
+
+## filter_confirmation Test 1:
+test_that("filter_confirmation Test 1: filter with first_cond", {
+  actual <-
+    filter_confirmation(
+      data,
+      by_vars = vars(USUBJID),
+      join_vars = vars(AVALC),
+      first_cond = AVALC == "CR" &
+        AVALC.join == "CR",
+      order = vars(AVISITN),
+      filter = TRUE
+    )
+
+  expected <- tibble::tribble(
+    ~USUBJID, ~AVISITN, ~AVALC,
+    "1",      2,        "CR"
+  )
+
+  expect_dfs_equal(
+    base = expected,
+    compare = actual,
+    keys = c("USUBJID", "AVISITN")
+  )
+})
+
+## filter_confirmation Test 3: filter with first_cond and summary function
+test_that("filter_confirmation Test 3: filter with first_cond and summary function", {
+  actual <-
+    filter_confirmation(
+      data,
+      by_vars = vars(USUBJID),
+      join_vars = vars(AVALC),
+      first_cond = AVALC == "PR" &
+        AVALC.join %in% c("CR", "PR"),
+      order = vars(AVISITN),
+      filter = count_vals(AVALC.join, "SD") <= 1
+    )
+
+  expected <- tibble::tribble(
+    ~USUBJID, ~AVISITN, ~AVALC,
+    "1",      1,        "PR"
+  )
+
+  expect_dfs_equal(
+    base = expected,
+    compare = actual,
+    keys = c("USUBJID", "AVISITN")
+  )
+})
+
+data <- tibble::tribble(
+  ~USUBJID, ~AVISITN, ~AVALC,
+  "1",      1,        "Y",
+  "1",      2,        "N",
+  "1",      3,        "Y",
+  "1",      4,        "N",
+  "2",      1,        "Y",
+  "2",      2,        "N",
+  "3",      1,        "Y",
+  "4",      1,        "N",
+  "4",      2,        "N",
+)
+
+filter_confirmation(
+  data,
+  by_vars = vars(USUBJID),
+  join_vars = vars(AVALC, AVISITN),
+  order = vars(AVISITN),
+  filter = AVALC == "Y" & AVALC.join == "Y" & AVISITN < AVISITN.join
+)
+
+data <- tibble::tribble(
+  ~USUBJID, ~AVISITN, ~AVALC,
+  "1",      1,        "PR",
+  "1",      2,        "CR",
+  "1",      3,        "NE",
+  "1",      4,        "CR",
+  "1",      5,        "NE",
+  "2",      1,        "CR",
+  "2",      2,        "PR",
+  "2",      3,        "CR",
+  "3",      1,        "CR",
+  "4",      1,        "CR",
+  "4",      2,        "NE",
+  "4",      3,        "NE",
+  "4",      4,        "CR",
+  "4",      5,        "PR"
+)
+
+filter_confirmation(
+  data,
+  by_vars = vars(USUBJID),
+  join_vars = vars(AVALC),
+  order = vars(AVISITN),
+  first_cond = AVALC.join == "CR",
+  filter = AVALC == "CR" & all(AVALC.join %in% c("CR", "NE")) &
+    count_vals(var = AVALC.join, val = "NE") <= 1
+)
```

Comment:
Yes, `derive_param_confirmed_bor()` calls `filter_confirmation()` separately for potential CR or PR responses. For other responses like SD `derive_param_confirmed_bor()` calls `filter()` because they do not depend on subsequent observations. See https://github.com/pharmaverse/admiralonco/blob/24_derive_param_confirmed_bor/R/derive_param_confirmed_bor.R.